src/hyp_map.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2022 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use page_tracking::{HwMemMap, HwMemRegion, HwMemRegionType, HwReservedMemType, HypPageAlloc};
+use riscv_page_tables::{FirstStagePageTable, PteFieldBits, PteLeafPerms, Sv48};
+use riscv_pages::{InternalClean, Page, PageAddr, PageSize, RawAddr, SupervisorPhys};
+use riscv_regs::{satp, LocalRegisterCopy, SatpHelpers};
+
+// Returns the base, size, and permission pair for the given region if that region type should be
+// mapped in the hypervisor's virtual address space.
+fn hyp_map_params(r: &HwMemRegion) -> Option<(PageAddr<SupervisorPhys>, u64, PteLeafPerms)> {
+    match r.region_type() {
+        HwMemRegionType::Available => {
+            // map available memory as rwx - unser what it'll be used for.
+            Some((r.base(), r.size(), PteLeafPerms::RWX))
```

Comment:
(Not related to your patch and no action requested, but we need to fix these permissions. Unused, random memory should _not_ have X perms.)

Reply:
Yeah, having hypervisor image as RWX is a bit scary too. I need to look up at how salus is loaded.

Reply:
I think @rajnesh-kanwal is doing this as part of his changes.